nautobot/extras/models/models.py

```diff
@@ -847,6 +849,16 @@ def enqueue_job(cls, func, name, obj_type, user, celery_kwargs=None, *args, sche
         if celery_kwargs is None:
             celery_kwargs = {}
 
+        job = get_job(name)
+        if job is not None:
+            try:
+                if getattr(job.Meta, "soft_time_limit", ""):
+                    celery_kwargs["soft_time_limit"] = job.Meta.soft_time_limit
+                if getattr(job.Meta, "time_limit", ""):
+                    celery_kwargs["time_limit"] = job.Meta.time_limit
+            except AttributeError:
+                pass
+
```

Comment:
Might be good as a sanity measure to compare `soft_time_limit` (if set) / `CELERY_TASK_SOFT_TIME_LIMIT` (otherwise) against `time_limit` (if set) / `CELERY_TASK_TIME_LIMIT` (otherwise) and warn the user if the hard limit is less than or equal to the soft limit?

Reply:
Good point. I'll write in some validation code